pkg/image/decompress.go

```diff
@@ -2,14 +2,94 @@ package image
 
 import (
 	"archive/tar"
-	"fmt"
+	"compress/gzip"
 	"io"
-	"os"
 	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
 )
 
-func IsCompressed(srcFile string) bool {
-	fExt := filepath.Ext(srcFile)
+var SupportedCompressionExtensions = []string{
+	Gz,
+}
+
+var SupportedArchiveExtentions = []string{
+	TarArch, TarArch + Gz,
+}
+
+// TODO fixup comment
+// UnpackData combines the decompressing and unarchiving of a data stream and returns the
+// end of the stream for further processing.
+func UnpackData(filename string, src io.Reader) interface{} {
+	glog.Infof("UnpackData: filename: %v, src: %T\n", filename, src)
+	/*if file is raw image{
+		return io.ReadCloser (raw image stream)
+	}*/
+	src = DecompressData(filename, src).(io.ReadCloser)
+	src = DearchiveData(filename, src).(io.Reader)
```

Comment:
So casting is need here only if `DecompressData` finds that the file is compressed and thus returns a ReadCloser, right?

Reply:
DearchibeData returns an `interface{}` which is then cast to io.Reader